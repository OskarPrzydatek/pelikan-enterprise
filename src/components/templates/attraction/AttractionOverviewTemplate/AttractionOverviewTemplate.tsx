import React from 'react';

import { Text } from '~/components/atoms';
import { Modal, OverviewListItem } from '~/components/molecules';
import { OverviewList, ErrorBundaryLoader } from '~/components/organisms';
import { useDeleteWithModal } from '~/hooks';
import { IAttractionData, IOverviewTemplate } from '~/models';

interface IAttractionOverviewTemplate extends IOverviewTemplate {
  data: IAttractionData[] | undefined;
}

export const AttractionOverviewTemplate: React.FC<
  IAttractionOverviewTemplate
> = ({
  data,
  page,
  isLoading,
  error,
  onClickEdit,
  onClickDelete,
  onClickNavigateToCreatePage,
}: IAttractionOverviewTemplate) => {
  const { isArray } = Array;
  const {
    showModal,
    handleItemIdBeforeDelete,
    onClickAcceptedDelete,
    onClickCloseModale,
  } = useDeleteWithModal(onClickDelete);

  return (
    <ErrorBundaryLoader error={error} isLoading={isLoading}>
      <OverviewList
        navigateLabel="Dodaj atrakcję"
        noItemsLabel="Brak atrakcji w systemie"
        page={page}
        title="Atrakcje"
        onClickNavigateToCreatePage={onClickNavigateToCreatePage}
      >
        {isArray(data)
          ? data.map(({ id, name }) => (
              <OverviewListItem
                key={`${id}-${name}`}
                dataTestID={`attraction-overview-list-item-${id}`}
                deleteIconTestID={`attraction-overview-delete-${id}`}
                editIconTestID={`attraction-overview-edit-${id}`}
                id={id}
                name={name}
                onClickDelete={handleItemIdBeforeDelete}
                onClickEdit={onClickEdit}
              />
            ))
          : null}
      </OverviewList>
      {showModal ? (
        <Modal
          acceptLabel="Usuń!"
          onClickAccept={onClickAcceptedDelete}
          onClickClose={onClickCloseModale}
        >
          <Text>Czy napewno usunąć element z listy?</Text>
        </Modal>
      ) : null}
    </ErrorBundaryLoader>
  );
};
