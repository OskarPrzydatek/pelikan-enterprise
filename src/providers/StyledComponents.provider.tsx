import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
