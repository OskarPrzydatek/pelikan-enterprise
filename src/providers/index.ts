export { StyledComponentsProvider } from './StyledComponents.provider';
