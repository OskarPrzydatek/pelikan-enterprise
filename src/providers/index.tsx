import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
