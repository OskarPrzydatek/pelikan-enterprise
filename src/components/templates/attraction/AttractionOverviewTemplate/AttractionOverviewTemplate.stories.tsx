import { Meta, StoryObj } from '@storybook/react';

import { attractionDataMock } from '~/mocks';
import { StyledComponentsProvider } from '~/providers';

import { AttractionOverviewTemplate } from './AttractionOverviewTemplate';

const meta: Meta<typeof AttractionOverviewTemplate> = {
  title: 'Templates/AttractionOverviewTemplate',
  component: AttractionOverviewTemplate,
};

export default meta;
type Story = StoryObj<typeof AttractionOverviewTemplate>;

export const AttractionOverviewTemplatePrimmary: Story = {
  render: () => {
    return (
      <StyledComponentsProvider>
        <AttractionOverviewTemplate
          data={attractionDataMock}
          error={undefined}
          isLoading={false}
          onClickDelete={() => console.log('onClickDelete!')}
          onClickEdit={() => console.log('onClickEdit!')}
          onClickNavigateToCreatePage={() =>
            console.log('onClickNavigateToCreatePage!')
          }
        />
      </StyledComponentsProvider>
    );
  },
};
