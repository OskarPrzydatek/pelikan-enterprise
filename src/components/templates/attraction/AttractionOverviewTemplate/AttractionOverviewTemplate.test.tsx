import { render } from '@testing-library/react';
import { describe, test, vi } from 'vitest';

import { StyledComponentsProvider } from '~/providers';

import { AttractionOverviewTemplate } from './AttractionOverviewTemplate';

const mockOnClickItem = vi.fn();
const mockOnClickNavigate = vi.fn();

describe('AttractionOverviewTemplate', () => {
  test('component snapshot', () => {
    const view = render(
      <StyledComponentsProvider>
        <AttractionOverviewTemplate
          items={[
            { id: 1, name: 'attraction1' },
            { id: 2, name: 'attraction2' },
            { id: 3, name: 'attraction3' },
            { id: 4, name: 'attraction4' },
            { id: 5, name: 'attraction5' },
          ]}
          onClickItem={mockOnClickItem}
          onClickNavigate={mockOnClickNavigate}
        />
      </StyledComponentsProvider>
    );
    expect(view).toMatchSnapshot();
  });
});
