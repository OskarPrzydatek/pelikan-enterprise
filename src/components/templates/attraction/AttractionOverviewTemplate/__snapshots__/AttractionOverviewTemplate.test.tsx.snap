// Vitest Snapshot v1

exports[`AttractionOverviewTemplate > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ilhmMj dgoPhb"
      >
        <div
          class="sc-eeMvmM bRaIPK"
        >
          <h2
            class="sc-jSUZER fTaKew"
          >
            Atrakcje
          </h2>
          <ul
            class="sc-cUEOzv lbvGZW"
          >
            <li
              class="sc-hBxehG bhOMdq"
              data-testid="attraction-overview-list-item-1"
            >
              <p
                class="sc-jrcTuL iBdZHI"
              >
                AttractiONe
              </p>
              <div
                class="sc-fnGiBr fhhnoL"
              >
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-edit-1"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-delete-1"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="sc-hBxehG bhOMdq"
              data-testid="attraction-overview-list-item-2"
            >
              <p
                class="sc-jrcTuL iBdZHI"
              >
                AdventuraLand
              </p>
              <div
                class="sc-fnGiBr fhhnoL"
              >
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-edit-2"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-delete-2"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="sc-hBxehG bhOMdq"
              data-testid="attraction-overview-list-item-3"
            >
              <p
                class="sc-jrcTuL iBdZHI"
              >
                WonderWorld
              </p>
              <div
                class="sc-fnGiBr fhhnoL"
              >
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-edit-3"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-delete-3"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="sc-hBxehG bhOMdq"
              data-testid="attraction-overview-list-item-4"
            >
              <p
                class="sc-jrcTuL iBdZHI"
              >
                ExoticEscape
              </p>
              <div
                class="sc-fnGiBr fhhnoL"
              >
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-edit-4"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-delete-4"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="sc-hBxehG bhOMdq"
              data-testid="attraction-overview-list-item-5"
            >
              <p
                class="sc-jrcTuL iBdZHI"
              >
                HistoryJourney
              </p>
              <div
                class="sc-fnGiBr fhhnoL"
              >
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-edit-5"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-gKPRtg dkRzHx"
                  data-testid="attraction-overview-delete-5"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
          </ul>
          <div
            class="sc-jOiSOi hXLofh"
          >
            <button
              class="sc-gKPRtg cylfAz"
            >
              <svg
                data-testid="arrow-icon"
                fill="none"
                viewBox="0 0 40 26"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                  fill="#013059"
                />
              </svg>
            </button>
            <p
              class="sc-jrcTuL iBdZHI"
              data-testid="pagination-counter"
            >
              1
               / 
              1
            </p>
            <button
              class="sc-gKPRtg gagpAz"
            >
              <svg
                data-testid="arrow-icon"
                fill="none"
                viewBox="0 0 40 26"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                  fill="#013059"
                />
              </svg>
            </button>
          </div>
          <button
            class="sc-eDvSVe fgoBjS"
            data-testid="overview-list-navigate-to-create-page"
          >
            Dodaj atrakcję
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ilhmMj dgoPhb"
    >
      <div
        class="sc-eeMvmM bRaIPK"
      >
        <h2
          class="sc-jSUZER fTaKew"
        >
          Atrakcje
        </h2>
        <ul
          class="sc-cUEOzv lbvGZW"
        >
          <li
            class="sc-hBxehG bhOMdq"
            data-testid="attraction-overview-list-item-1"
          >
            <p
              class="sc-jrcTuL iBdZHI"
            >
              AttractiONe
            </p>
            <div
              class="sc-fnGiBr fhhnoL"
            >
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-edit-1"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-delete-1"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-hBxehG bhOMdq"
            data-testid="attraction-overview-list-item-2"
          >
            <p
              class="sc-jrcTuL iBdZHI"
            >
              AdventuraLand
            </p>
            <div
              class="sc-fnGiBr fhhnoL"
            >
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-edit-2"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-delete-2"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-hBxehG bhOMdq"
            data-testid="attraction-overview-list-item-3"
          >
            <p
              class="sc-jrcTuL iBdZHI"
            >
              WonderWorld
            </p>
            <div
              class="sc-fnGiBr fhhnoL"
            >
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-edit-3"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-delete-3"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-hBxehG bhOMdq"
            data-testid="attraction-overview-list-item-4"
          >
            <p
              class="sc-jrcTuL iBdZHI"
            >
              ExoticEscape
            </p>
            <div
              class="sc-fnGiBr fhhnoL"
            >
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-edit-4"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-delete-4"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-hBxehG bhOMdq"
            data-testid="attraction-overview-list-item-5"
          >
            <p
              class="sc-jrcTuL iBdZHI"
            >
              HistoryJourney
            </p>
            <div
              class="sc-fnGiBr fhhnoL"
            >
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-edit-5"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-gKPRtg dkRzHx"
                data-testid="attraction-overview-delete-5"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
        </ul>
        <div
          class="sc-jOiSOi hXLofh"
        >
          <button
            class="sc-gKPRtg cylfAz"
          >
            <svg
              data-testid="arrow-icon"
              fill="none"
              viewBox="0 0 40 26"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                fill="#013059"
              />
            </svg>
          </button>
          <p
            class="sc-jrcTuL iBdZHI"
            data-testid="pagination-counter"
          >
            1
             / 
            1
          </p>
          <button
            class="sc-gKPRtg gagpAz"
          >
            <svg
              data-testid="arrow-icon"
              fill="none"
              viewBox="0 0 40 26"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                fill="#013059"
              />
            </svg>
          </button>
        </div>
        <button
          class="sc-eDvSVe fgoBjS"
          data-testid="overview-list-navigate-to-create-page"
        >
          Dodaj atrakcję
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
